### PR TITLE
Service 02

### PR DESCRIPTION
**Pull request by michalkocon**

### CODE REVIEW

File: asn1-pusc-lib.pro [hunk 1]
```diff
@@ -65,7 +65,38 @@ DISTFILES += \
...
+    service-02/PUS-2-5.acn \
+    service-02/PUS-2-5.asn1 \
+    service-02/PUS-2-6.acn \
+    service-02/PUS-2-6.asn1 \
+    service-02/PUS-2-7.acn \
+    service-02/PUS-2-7.asn1 \
+    service-02/PUS-2-8.acn \
+    service-02/PUS-2-8.asn1 \
+    service-02/PUS-2-9.acn \
+    service-02/PUS-2-9.asn1
```
Comment (hcorg) [inline asn1-pusc-lib.pro]: maybe really it's time to either create script filling this file, or use some wildcards :)

File: service-02/LogicalDevice.acn [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+--! MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+--! GNU General Public License for more details.
+--!
+--! You should have received a copy of the GNU General Public License
+--! along with this program.  If not, see <http://www.gnu.org/licenses/>.
+--!
+
+LogicalDevice DEFINITIONS ::= BEGIN
+
+LogicalDevice-ID [size 32, encoding pos-int] --TODO
```
Comment (hcorg) [inline service-02/LogicalDevice.acn]: `encode-values`?

Comment (hcorg) [reply service-02/LogicalDevice.acn]: it's a general comment for all ENUMERATED

File: service-02/PUS-2-2.asn1 [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    PUSC-UINT32 FROM BasicTypes
+    Register FROM Registers;
+
+TC-2-2-DistributeRegisterCommands ::= SEQUENCE
+{
+--! n PUSC-UINT32, moved to acn
+    registerCommands SEQUENCE (SIZE(0 .. 2147483647)) OF Register
+}
+
+
```
Comment (hcorg) [inline service-02/PUS-2-2.asn1]: why so many `\n`?

File: service-02/PUS-2-12.asn1 [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+PUS-2-12 DEFINITIONS AUTOMATIC TAGS ::= BEGIN
+EXPORTS ALL;
+IMPORTS
+    Transaction-ID, TransactionExecutionStatus FROM Transaction;
+
+TM-2-12-LogicalDeviceDataReport ::= SEQUENCE
+{
+    transaction-ID Transaction-ID,
+    transactionExecutionStatus TransactionExecutionStatus,
+    parameterValue OCTET STRING (SIZE(4)) --TODO
```
Comment (hcorg) [inline service-02/PUS-2-12.asn1]: why TODO? what should be here?

File: asn1-pusc-lib.pro (hunk 1 above)
Comment (michalkocon) [reply asn1-pusc-lib.pro]: IDE is filling this ;)

File: service-02/PUS-2-12.asn1 (hunk 4 above)
Comment (michalkocon) [reply service-02/PUS-2-12.asn1]: It's dependent on transaction-ID, similarly  to TM-2-9 with transaction-ID and dataBlock

Comment (michalkocon) [reply service-02/PUS-2-12.asn1]: Maybe TODO is not the best way to describe it. It's dependent on transaction-ID, so it probably cannot be modeled within ASN.1/ACN. It's a dummy type, similarly to dataBlock in PUS-2-9

File: service-02/PUS-2-10.asn1 [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+EXPORTS ALL;
+IMPORTS
+    PUSC-UINT32 FROM BasicTypes
+    LogicalDevice-ID FROM LogicalDevice;
+
+maxTC-2-10-LogicalDeviceCommandCount INTEGER ::= 2147483647
+
+TC-2-10-DistributeLogicalDeviceCommands ::= SEQUENCE
+{
+--! n TUnsignedInteger, moved to acn
```
Comment (hcorg) [inline service-02/PUS-2-10.asn1:34]: TUnsignedInteger? maybe disregard all such comments in future ;) (comments lie :D )

**Review by hcorg — APPROVED** [review]

**Pull request merged**